rust/experimental/query_engine/kql-parser/src/tabular_expressions.rs

```diff
@@ -354,6 +416,9 @@ pub(crate) fn parse_tabular_expression(
                 }
             }
             Rule::where_expression => expressions.push(parse_where_expression(rule, state)?),
+            Rule::summarize_expression => {
+                expressions.push(parse_summarize_expression(rule, state)?)
+            }
             _ => panic!("Unexpected rule in tabular_expression: {rule}"),
```

Comment:
Same issue here. If this branch can never happen then you should add a comment describing why it's safe to use panic in this context.

Reply:
These should really never be hit unless a dev makes a huge mistake 🤣 We could convert them to errors though, no real harm in that. BUT. This pattern is all over KQL parser code. Would you mind clearing your block on this PR and I'll work on this as a follow-up for the whole crate? I don't want to explode the diff here for such a cleanup task.